### PR TITLE
Allow exclusions to the consistentDependencies rule

### PR DESCRIPTION
**Pull request by gluxon**
This PR adds an `ignoredDependencies` option to the `consistentDependencies` rule.

This option may seem unnecessary when consumers can simply remove the dependency from the root `dependencies` or `devDependencies` object. However there are some cases where you want to control which version of a dependency gets hoisted to the root `node_modules`, while still having different versions within each sub-package.

For example:

```
node_modules
  @types/foo <-- version 1.0 gets hoisted
  @types/bar <-- wants @types/foo 2.0, but sees 1.0

packages/
  a
    node_modules
      @types/foo <-- should be 1.0

  b
    node_modules
      @types/foo <-- should be 2.0
```

In this case, we want to specify `@types/foo` to be `2.0` at the root so a different version gets hoisted.

```json
{
  "devDependencies": {
    "@types/foo": "2.0"
  }
}
```

But this currently requires completely disabling the `consistentDependencies` check.

### CODE REVIEW

**Review by ericanderson — APPROVED** [review]

**Comment by gluxon** [comment]
Thanks @ericanderson! I'll wait for someone with write access to merge since I can't self-merge.

**Pull request merged**
